lib/SIL/SILDeclRef.cpp

```diff
@@ -772,6 +772,22 @@ std::string SILDeclRef::mangle(ManglingKind MKind) const {
   llvm_unreachable("bad entity kind!");
 }
 
+bool SILDeclRef::requiresNewVTableEntry() const {
+  if (cast<AbstractFunctionDecl>(getDecl())->needsNewVTableEntry())
+    return true;
+  if (kind == SILDeclRef::Kind::Allocator) {
```

Comment:
@jrose-apple I don't really understand what was going on here. Even if the initializer is `dynamic`, we still have a vtable entry if its `required`, which seems strange, but it lines up with what SILGenApply.cpp was already doing.

Reply:
If it's `dynamic` from the point where it's introduced then I guess we wouldn't need the vtable entry, but if it's marked `dynamic` but overrides another required initializer we'd need both.

Reply:
…right, that's already covered by the above check. Yeah, okay, `dynamic` shouldn't need this, but then we need to be careful about MissingMemberDecl not trying to add one back if the initializer can't be deserialized.